pkg/scheduler/schedule_one.go

```diff
@@ -309,12 +309,21 @@ func (sched *Scheduler) bindingCycle(
 		return status
 	}
 
-	// Any failures after this point cannot lead to the Pod being considered unschedulable.
+	// Any failures after this point cannot lead to the Pod being considered unschedulable and so we create the PodScheduled condition.
 	// We define the Pod as "unschedulable" only when Pods are rejected at specific extension points, and PreBind is the last one in the scheduling/binding cycle.
 	//
 	// We can call Done() here because
 	// we can free the cluster events stored in the scheduling queue sonner, which is worth for busy clusters memory consumption wise.
 	sched.SchedulingQueue.Done(assumedPod.UID)
+	podScheduled := &v1.PodCondition{
+		Type:   v1.PodScheduled,
+		Status: v1.ConditionTrue,
+	}
+	newStatus := assumedPod.Status.DeepCopy()
+	podutil.UpdatePodCondition(newStatus, podScheduled)
```

Comment:
I think you need to check the update result at here? Similiar to https://github.com/kubernetes/kubernetes/blob/1504f10e7946f95a8b1da35e28e4c7453ff62775/pkg/scheduler/framework/preemption/preemption.go#L181-L187